drivers/mmc/host/bcm2835-sdhost.c

```diff
@@ -815,7 +815,11 @@ static void bcm2835_sdhost_prepare_dma(struct bcm2835_host *host,
 			if (sg_is_last(sg)) {
 				BUG_ON(sg->length < len);
 				sg->length -= len;
+#ifdef CONFIG_ARM64
+				host->drain_page = sg_page(sg);
```

Comment:
Yes, I just didn't want to be responsible for testing all the different combinations.  The macro is very simple through if you want to look at it.